### ISSUE TITLE
Idea for warp fuel - crystals

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#5294]
Light cruise uses liquid fuel thus players will have to manage a fuel tank. But with warp, I’m concerned if it is also a liquid fuel, it may get complicated to manage two tanks. Warp fuel was a discussion long ago, but with gameplay elements being added, I’d like to revive the idea of an NMS style “single use per jump” fuel. But with some extra gameplay elements that intersect well with the whole manufacturing element.

Player’s mine “warp crystals” as a solid resource. Crystals would be a distinct resource type from normal ore visually. There would different rarities of crystals. Let’s stick with green (common), blue (uncommon), and purple (rare) for this example. Spawn/generation rates for each would match the rarity.
![image](https://github.com/tsunamayo/Starship-EVO/assets/36980446/d952a62a-0bc2-49f4-8b26-2e40cb6df422)
- Lego power crystal because it is the best crystal :D

These crystals would be fed though a factory block, call it a “Crystal Refiner”, the output of which would be warp fuel “canisters”. These you would plug into a warp core, ships computer, or maybe some place in the galaxy map that way you can easily access it. But that’s beside the point.

This is where the crystal rarity comes back. The rarity of the processed crystal determines the number of jumps for each canister. For example:

Green (Common): 4 jumps per.
Blue (Uncommon): 6 jumps per.
Purple (Rare): 8 jumps per.
![image](https://github.com/tsunamayo/Starship-EVO/assets/36980446/623cd69e-7644-4ab1-b7b1-3a4bcbf146a0)

The number of jumps remaining would be displayed on the ship status bar.
![image](https://github.com/tsunamayo/Starship-EVO/assets/36980446/1be0a4d5-d614-43ac-9331-e3952446f8b2)

The advantage of this system is that it is completely separate from “cruise fuel”, but simple to understand. It utilizes the existing factory system to have some degree of processing and automation. The multiple jumps per crystal reason was already stated. Different rarities add a new side to economy where rarer crystals sell for more. And most importantly, it gives a degree of advantage to those who find rarer crystals when mining but without diminishing the gameplay experience for those who don’t bother.

**Comment by EmceeAlpin** [comment]
I completely support this request/suggestion. Having a crystal fuel that focuses energy into the warp blocks makes a lot of sense. Something about warp blocks consuming liquid fuel felt cumbersome. I think this will be more straight forward, easier to understand, and to plan for.

**Comment by CoasterBlock** [comment]
I 100% agree, it feels more belivable than having to pump hydrogen into a device capable of FTL. Its like using a steamengine to power a particle collider. A type of crystal that  would need to be obtained/maintained would be a fantastic idea, plus it would perhaps make systems for survival more interesting by making FTL flight actually something that needs work to be achieved. Something that needs infrastructure on board of a ship to function.

**Comment by TIKIRobo** [comment]
This would also help with progression so that players have to get LC and then warp instead of both at the same time.
for instance a player could get LC so that they can get to the asteroids or moons of their starting system to get the warp crystals that would let them build/power their warp ships to let them move on to other systems

**Comment by Majikmonster** [comment]
I approve of this separation between fuel sources. It makes both progression and function more nuanced.

If there's going to be different types of crystal, I'd go for size rather than color, based on a similarity in the real world with ore chunks and gemstones. Larger chunks are more rare, found in veins and deposits, whereas smaller pieces are widely scattered and discovered by accident. This would prioritize prospecting for concentrations of crystals in later stages of gameplay when larger ships are travelling further and players/fleets have the investment of a mining op, while small ships and crew need only the bits to meet their needs and can scavenge what's easily acquired.

**Comment by tsunamayo** [comment]
@ZachZent Yep thanks I agree with what explained, this was more or less my plan.

1. I dont want to use gas, it does not make much sense and it seems a bit too easy
2. I always wanted to have the player to find / mine a solid ressources
3. Now it order to be meaningful, as resources are pretty much limitness have to find a way to artificially limit how much the player can carry. For the usual cargo/fuel it is simply a matter of volume and mass, but I dont want to have the warp resources to be big, it should look somewhat precious and exotic! This is why I need to introduce a special storage type that will require power to function, it that way the maximum number of warp a ship can do will be a trade-off with the other system at a given power level. 
This will also directly collide with the warp range: do you prefer a ship that can store more warp (use more warp storage) or that can warp further away (use more warp core)? 
As it is supposed to be somekind of exotic / anti-matter / dark matter dark energy or whatever we will come up with it make sense to require a special container.

I havent though about the rarity but I fancy the idea, it makes for more exciting discovery. I will need to think more about what kind of resources is needed, and whether it require some refining (most likely it wont).

Cheers

**Comment by tsunamayo** [comment]
> This would also help with progression so that players have to get LC and then warp instead of both at the same time. for instance a player could get LC so that they can get to the asteroids or moons of their starting system to get the warp crystals that would let them build/power their warp ships to let them move on to other systems

So yes it will be used for the early game player progression:
It will work like that (very rough view): craft first ship, craft h2 from water -> get in space and get mineral require for lightcruise from nearby asteroid -> get to another planet for warp fuel

**Comment by ZachZent** [comment]
Sounds great to me. But please go with crystals as the minable resource. Visually distinct and will look very cool floating in warp containers.

May I make a bit of a tutorial suggestion. Had this idea in my head.

Act 0: The KOTOR scene. Crew member on a ship under attack. Learn the basics of movement, combat, repair, etc. The hook (note, this can be skipped for...)

Act I: The NMS scene. Start on a ring - maybe make rings not have many natural resources including warp crystals as it is an artificial station - next to a broken down escape pod/ship.Cannibalize it to make a vehicle (hover or wheels?) - introduces to basic crafting and building. After completing the vehicle, the player is directed to a nearby town up ring.

Totally fandom note. The tutorial should be guided by FR-3N. This also allows you to add in personal drones for players, maybe something that players can modify to their tastes. But that's getting into a tangent.
![image](https://github.com/tsunamayo/Starship-EVO/assets/36980446/39799e6a-0af9-498c-a7d2-b4e60ac391d3)

Act II: The town scene. Arriving at the town, players can do some simple quests that teach different aspects of the game and earn some cash. Like some basic combat, mining, exploration. Get enough money to buy some parts of a fighter and build the rest from materials.

Act III: Everything else. Fly up to get cruise fuel, to another planet, a nearby station. Etc.

**Comment by tsunamayo** [comment]
@ZachZent Unfortunately I am totally not doing this. 
First It would take me forever to code, but most importantly be very likely to kill the game: it is very complicated for me to test all the weird stuff newcomers can do, they give up as soon as they see a bug or something hard to understand, not reporting their issue leaving in the blind with very little information on where the problem is.
I have learned that the hard way with the current tutorial, it was already too complicated.

The first few hours are too critical, so unfortunately it will be extremely basic. This is why I have added the help section, it will be mostly about exploring that help to learn the fundamentals.
